pkg/controllers/grafana-operator/controller.go

```diff
@@ -166,6 +160,31 @@ func RegisterWithManager(mgr ctrl.Manager) error {
 		return err
 	}
 
+	go func() {
+		stop := make(chan struct{})
+		wait.Until(func() {
+			logger.V(6).Info("Trying to establish a watch on Grafana resources")
+			var datasources integreatlyv1alpha1.GrafanaDataSourceList
+			if err := r.k8sClient.List(context.Background(), &datasources, client.InNamespace("default")); err != nil {
+				logger.V(6).Info("GrafanaDataSource CRD does not exist", "err", err)
+				return
+			}
+			grafanaInformer := r.grafanaInformer()
+			go grafanaInformer.Run(nil)
+			if err := c.Watch(&source.Informer{
+				Informer: grafanaInformer,
+			}, &handler.EnqueueRequestForObject{}, predicate.GenerationChangedPredicate{}); err != nil {
+				logger.Error(err, "Could not establish a watch on Grafana CRs")
```

Comment:
if the watch couldn't be established wouldn't this result in just logging an error and never attempting to watch again? 

How about we add the watch dynamically in the reconcile loop? 
e.g. see: https://github.com/rhobs/monitoring-stack-operator/pull/100/files#diff-f7ca2b110a0e9c32221f61f0b87534e8cf46132d81b5ce267e5679396d969c8fR158